rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java

```diff
@@ -956,33 +992,1829 @@ void annotationWrappingWithNulls() {
+            spaces -> spaces,
+            wrap -> wrap.withArrayInitializer(wrap.getArrayInitializer().withWrap(WrapAlways).withAlignWhenMultiline(true))
+          ),
+          java(
+            """
+              class Test {
+                  int[] numbers = {1, 2, 3, 4, 5};
+              }
+              """,
+            """
+              class Test {
+                  int[] numbers = {1,
+                                   2,
+                                   3,
+                                   4,
+                                   5};
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void arrayInitializerNewLineAfterOpeningCurly() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withArrayInitializer(wrap.getArrayInitializer().withWrap(WrapAlways).withNewLineAfterOpeningCurly(true))
+          ),
+          java(
+            """
+              class Test {
+                  int[] numbers = {1, 2, 3, 4, 5};
+              }
+              """,
+            """
+              class Test {
+                  int[] numbers = {
+                          1,
+                          2,
+                          3,
+                          4,
+                          5};
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void arrayInitializerPlaceClosingCurlyOnNewLine() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withArrayInitializer(wrap.getArrayInitializer().withWrap(WrapAlways).withPlaceClosingCurlyOnNewLine(true))
+          ),
+          java(
+            """
+              class Test {
+                  int[] numbers = {1, 2, 3, 4, 5};
+              }
+              """,
+            """
+              class Test {
+                  int[] numbers = {1,
+                          2,
+                          3,
+                          4,
+                          5
+                  };
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void arrayInitializerAllOptions() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withArrayInitializer(wrap.getArrayInitializer()
+              .withWrap(WrapAlways)
+              .withNewLineAfterOpeningCurly(true)
+              .withPlaceClosingCurlyOnNewLine(true))
+          ),
+          java(
+            """
+              class Test {
+                  int[] numbers = {1, 2, 3, 4, 5};
+              }
+              """,
+            """
+              class Test {
+                  int[] numbers = {
+                          1,
+                          2,
+                          3,
+                          4,
+                          5
+                  };
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void arrayInitializerDoNotWrap() {
```

Comment:
Elsewhere we seem to use
```suggestion
    void arrayInitializerDoNotWrapAlreadyCorrect() {
```
Or would we want to add a before text block here that's not already as we want it?